cdserver/app/models/floor.rb

```diff
@@ -3,6 +3,7 @@ class Floor < ApplicationRecord
   has_many :places, as: :compartment
   belongs_to :shelter_building
   belongs_to :case_type
+  belongs_to :program_type
```

Comment:
I believe program_type should be inheritable from shelter_building the same way case_type is.   It is possible the entire building could be devoted to "General"  In fact that is the typical case for Families with Children.